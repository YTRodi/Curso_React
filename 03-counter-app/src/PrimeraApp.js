import React from 'react';
import PropTypes from "prop-types";

// Functional Components ( Regresa un component de HTML )
const PrimeraApp = ( { saludo, subtitulo } ) => {
    
    return (

        <>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>            
        </>
       
    ); 

};

// Si alguien quiere usar este componente, tiene que proveer la propiedad 'saludo'.
PrimeraApp.propTypes = {

    // Le digo que el saludo es un string y es obligatorio
    saludo: PropTypes.string.isRequired

}

// Si queremos que algunas propiedades traigan por default
PrimeraApp.defaultProps = {

    subtitulo: 'Soy un subtitulo'

}


export default PrimeraApp;