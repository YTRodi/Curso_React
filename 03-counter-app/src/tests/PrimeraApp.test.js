import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en el componente < PrimeraApp />', () => {
    
    // Pruebas con enzyme
    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
        
        const saludoTest = 'Ohayo';

        // shallow es parecido al render, pero nos permite simular clicks, document.queryselector y otras cosas..
        const wrapper = shallow( <PrimeraApp saludo={ saludoTest }/> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        
        const saludoTest = 'Ohayo';
        const subtituloTest = 'Soy un subtitulo';

        const wrapper = shallow(
             <PrimeraApp 
                saludo = { saludoTest }
                subtitulo = { subtituloTest }
             /> 
        );

        // find() = document.querySelector('#id' o '.class' o etiquetas en general);
        const textoParrafo = wrapper.find( 'p' ).text();
        // console.log(textoParrafo);


        expect( textoParrafo ).toBe( subtituloTest );

        expect( wrapper ).toMatchSnapshot();

    });

});