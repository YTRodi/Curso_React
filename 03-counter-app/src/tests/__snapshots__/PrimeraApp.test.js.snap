// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente < PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Ohayo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en el componente < PrimeraApp /> Debe de mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Ohayo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
